### PR TITLE
01-07 bugs and minor changes

### PR DESCRIPTION
**Pull request by jonathclee**
Bug fixes and minor changes to notebook 01 - 07.

Most of content remains unchanged. Most bugs are subtle, relating to data downloading and library dependencies. Notebook 01-07 can now be run on Google Colab without issue (as of 14/8/2025).

One moderate change is the about the sequence of train test split and data normalisation. The original code normalised the data first then train test split, e.g. 

```Python
X = df[feature_names].values
X = StandardScaler().fit_transform(X)
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, ...)
# ... do the model training ...
```

But it is better to first train test split then normalise the data. Otherwise, data leakage could happen (test set can somehow be "seen" by the model). It is a known problem / common practice in ML communities I think. Hence, modify as follows.

```Python
X = df[feature_names].values
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, ...)
scalar = StandardScaler()
X_train = scalar.fit_transform(X_train)
X_test = scalar.transform(X_test)     # dont fit test set, only transform using fitted scalar
# ... do the model training ...
```
Though the results / performance does not differ much.

Briefly, here is the changes to 01-07 notebook:

- `01_recap_python.ipynb`: Fixed bugs with data loading and package dependencies (`xarray[io]`, `cftime`). Added Google drive mounting as well.
- `02_model_train_evaluation.ipynb`: No issue.
- `03_linear_models_dim_reduction.ipynb`: Fixed bugs with image plotting in the last cell. Related to images not being downloaded remotely.
- `04_clustering.ipynb`: Can run from fresh. Add other metrics/scores for clustering.
- `05_classification.ipynb`: Can run from fresh. Changed the sequence of train test split and data normalisation (see above).
- `06_trees.ipynb`: Can run from fresh. Changed the sequence of train test split and data normalisation (see above).
- `07_forests.ipynb`: No bug and can run from fresh.. Add `xgboost` exercise.

See if any change is needed. Can modify accordingly.

### CODE REVIEW

**Comment by jonathclee** [comment]
Prevent some metadata being committed (Still some persists, don't know why)
